### PR TITLE
Follow-up backend 1.4 release prep

### PR DESCRIPTION
**Pull request by phschiele**
## Description
Small follow-up to #2239.

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/lin_ops/canon_backend.py [hunk 1]
```diff
@@ -1056,7 +1056,7 @@ def mul(self, lin: LinOp, view: SciPyTensorView) -> SciPyTensorView:
         if is_param_free_lhs:
             reps = view.rows // lhs.shape[-1]
             if reps > 1:
-                stacked_lhs = (sp.kron(sp.eye(reps, format="csc"), lhs))
+                stacked_lhs = (sp.kron(sp.eye(reps, format="csr"), lhs))
```
Comment (phschiele) [inline cvxpy/lin_ops/canon_backend.py:1059]: Here we are in the param-free branch, so we should use CSR.

File: cvxpy/lin_ops/canon_backend.py [hunk 2]
```diff
@@ -1209,7 +1209,7 @@ def func(x, p):
             else:
                 new_rows = x_row * (rows + 1) - k
             new_rows = (new_rows + x_slice * total_rows).astype(int)
-            return sp.csr_matrix((x.data, (new_rows, x.col)), shape)
+            return sp.csc_matrix((x.data, (new_rows, x.col)), shape)
```
Comment (phschiele) [inline cvxpy/lin_ops/canon_backend.py:1212]: This could be a parametrized expression, so we should use CSC.

**Review by PTNobel — APPROVED** [review]
This seems like mostly cleaning up sparse matrix formats and should all be correct

**Comment by phschiele** [comment]
> This seems like mostly cleaning up sparse matrix formats and should all be correct

Some intuition:
We prefer CSR for fast row slicing. However, having parameters leads to **many** zero rows, so CSR becomes impractical / prohibitive. 

E.g., a `Variable(3)` looks like 
```
1 0 0
0 1 0
0 0 1
```
in the backend, but a `cp.multiply(Variable(3), Parameter(3))` looks like
```
1 0 0
0 0 0
0 0 0
0 0 0
0 1 0
0 0 0
0 0 0 
0 0 0
0 0 1
```
I.e., the number of rows is roughly $\mathcal{O}(n * p)$ (where n is the number of variables and p the number of parameters), but the nonzero rows is $\mathcal{O}(n)$. We often even have $n \approx p$.

**Pull request merged**